### DISCUSSION TITLE
No Mapeamento de datas

### DISCUSSION THREAD

**Discussion by joshuajka** [balta-io/forum#444] [Tira Dúvidas] [answered]
No Mapeamento de datas posso usar o tipo DateTime ou melhor o SmalDateTime?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @joshuajka, como vai?

O SQL Server suporta três (Até onde sei) tipos de data/hora. O DATETIME é o primeiro e mais antigo, que suporta um longo período de data.
https://learn.microsoft.com/pt-br/sql/t-sql/data-types/datetime-transact-sql?view=sql-server-ver16

O SMALLDATETIME suporta um período bem menor de data, porém é mais otimizado.
https://learn.microsoft.com/en-us/sql/t-sql/data-types/smalldatetime-transact-sql?view=sql-server-ver16

O DATETIME2 é a atual recomendação padrão para data/hora, similar ao DATETIME mas com algumas otimizações.
https://learn.microsoft.com/pt-br/sql/t-sql/data-types/datetime2-transact-sql?view=sql-server-ver16

Sobre sua pergunta, a resposta depende de qual data você vai armazenar. Se for algo que caiba no tamanho do SmallDateTime, pode usar ele sim.
